From 7577a4de51ee98325ceb10e023c1aaa5fb7d3390 Mon Sep 17 00:00:00 2001
From: Daniel Tameling <tamelingdaniel@gmail.com>
Date: Mon, 21 May 2018 16:53:31 +0200
Subject: [PATCH 4/4] don't leave pager if prev/next-entry on first/last
 message

other operations also need fixing. would be better if leaving index
before looking at operations would be fixed instead of fixing every
single operation.
---
 index.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/index.c b/index.c
index 8cc92fcc0..030b516d2 100644
--- a/index.c
+++ b/index.c
@@ -2529,7 +2529,12 @@ int mutt_index_menu(void)
         if (menu->current >= (Context->mailbox->vcount - 1))
         {
           if (menu->menu == MENU_MAIN)
-            mutt_message(_("You are on the last message"));
+            mutt_error(_("You are on the last message."));
+          else if (menu->menu == MENU_PAGER)
+          {
+            op = OP_DISPLAY_MESSAGE;
+            continue;
+          }
           break;
         }
         menu->current++;
@@ -2572,7 +2577,12 @@ int mutt_index_menu(void)
         if (menu->current < 1)
         {
           if (menu->menu == MENU_MAIN)
-            mutt_message(_("You are on the first message"));
+            mutt_error(_("You are on the first message."));
+          else if (menu->menu == MENU_PAGER)
+          {
+            op = OP_DISPLAY_MESSAGE;
+            continue;
+          }
           break;
         }
         menu->current--;
-- 
2.20.1 (Apple Git-117)

